homeassistant/core.py

```diff
@@ -1257,6 +1265,100 @@ def as_dict(self) -> Dict:
             'config_source': self.config_source
         }
 
+    def set_time_zone(self, time_zone_str: Optional[str]) -> None:
+        """Help to set the time zone."""
+        if time_zone_str is None:
+            return
+
+        time_zone = dt_util.get_time_zone(time_zone_str)
+
+        if time_zone:
+            self.time_zone = time_zone
+            dt_util.set_default_time_zone(time_zone)
+        else:
+            _LOGGER.error("Received invalid time zone %s", time_zone_str)
+
+    @callback
+    def _update(self,
+                source: str,
+                latitude: Optional[float] = None,
+                longitude: Optional[float] = None,
+                elevation: Optional[int] = None,
+                unit_system: Optional[str] = None,
+                location_name: Optional[str] = None,
+                time_zone: Optional[str] = None) -> None:
+        """Update the configuration from a dictionary.
+
+        Async friendly.
+        """
+        self.config_source = source
+        if latitude is not None:
+            self.latitude = latitude
+        if longitude is not None:
+            self.longitude = longitude
+        if elevation is not None:
+            self.elevation = elevation
+        if unit_system is not None:
+            if unit_system == CONF_UNIT_SYSTEM_IMPERIAL:
+                self.units = IMPERIAL_SYSTEM
+            else:
+                self.units = METRIC_SYSTEM
+        if location_name is not None:
+            self.location_name = location_name
+        if time_zone is not None:
+            self.set_time_zone(time_zone)
+
+    async def update(self,
+                     latitude: Optional[float] = None,
+                     longitude: Optional[float] = None,
+                     elevation: Optional[int] = None,
+                     unit_system: Optional[str] = None,
+                     location_name: Optional[str] = None,
+                     time_zone: Optional[str] = None) -> None:
+        """Update the configuration from a dictionary.
+
+        Async friendly.
+        """
+        from homeassistant.config import SOURCE_STORAGE
```

Comment:
Let's store the constant either in const or at top of this file

Reply:
Fixed